packages/cli-core/src/printer/printer.ts

```diff
@@ -37,16 +72,152 @@ export class Printer {
    */
   static print = (message: string, colorName?: COLOR) => {
     if (colorName) {
-      console.log(color(colorName, message));
+      this.writeStream.write(color(colorName, message));
     } else {
-      console.log(message);
+      this.writeStream.write(message);
     }
   };
 
+  /**
+   * Logs a message with animated ellipsis
+   */
+  static async indicateProgress(
+    message: string,
+    callback: () => Promise<void>
+  ) {
+    try {
+      this.instance.startAnimatingEllipsis(message);
+      await callback();
+    } finally {
+      this.instance.stopAnimatingEllipsis(message);
+    }
+  }
+
   /**
    * Prints a new line to console
    */
   static printNewLine = () => {
-    console.log(EOL);
+    this.writeStream.write(EOL);
   };
+
+  /**
+   * Logs a message to the console.
+   */
+  static log(message: string, level: LogLevel = LogLevel.INFO, eol = true) {
```

Comment:
we should have _either_ `log(message, level)` _or_ `info(message)`, `debug(message)` but not both. I think I have a slight preference for the former because it's fewer methods on the class, but I don't have a strong opinion one way or the other. @sobolk any opinion here?

Reply:
All the loggers I know have both APIs. See
https://github.com/winstonjs/winston?tab=readme-ov-file#using-the-default-logger
https://logging.apache.org/log4j/2.x/javadoc/log4j-api/org/apache/logging/log4j/Logger.html
https://learn.microsoft.com/en-us/aspnet/core/fundamentals/logging/?view=aspnetcore-8.0

Having said that, doing one patter and introducing other later is always an option. 
But my sophisticated algorithm chose `log` so I'd be fine with that first.
![image](https://github.com/aws-amplify/amplify-backend/assets/5849952/3825de9e-04d1-4d07-8be8-8b291d4e7e99)